From 2e7cb4e8bf96eded4d57b8ff43152166bad9bc38 Mon Sep 17 00:00:00 2001
From: iberezovskiy <iberezovskiy@mirantis.com>
Date: Wed, 2 Dec 2015 17:28:27 +0300
Subject: [PATCH] Follow-up on PyMySQL support for Red Hat platforms
 (Keystone).

Rely on packaging dependencies to avoid issues caused by different
package names between Fedora and RHEL (python-PyMySQL vs python2-PyMySQL).

https://review.openstack.org/#/c/245229/4/spec/classes/neutron_db_spec.rb
includes all the discussion that led to this.

Change-Id: Iff047fab81f620f8df5a40296d23203461949546
(cherry picked from commit dd72e6d5495c2cedddb3d0da027955f8ce8b0b89)
---
 keystone/manifests/params.pp              |  2 +-
 keystone/spec/classes/keystone_db_spec.rb | 31 +++++++++++++++++++++----------
 2 files changed, 22 insertions(+), 11 deletions(-)

diff --git a/keystone/manifests/params.pp b/keystone/manifests/params.pp
index 0c8cc9b..d8c4737 100644
--- a/keystone/manifests/params.pp
+++ b/keystone/manifests/params.pp
@@ -32,7 +32,7 @@ class keystone::params {
       $service_provider             = undef
       $keystone_wsgi_script_source  = '/usr/share/keystone/keystone.wsgi'
       $paste_config                 = '/usr/share/keystone/keystone-dist-paste.ini'
-      $pymysql_package_name         = 'python2-PyMySQL'
+      $pymysql_package_name         = undef
     }
   }
 }
diff --git a/keystone/spec/classes/keystone_db_spec.rb b/keystone/spec/classes/keystone_db_spec.rb
index 1918c95..2ff4213 100644
--- a/keystone/spec/classes/keystone_db_spec.rb
+++ b/keystone/spec/classes/keystone_db_spec.rb
@@ -34,8 +34,6 @@ describe 'keystone::db' do
       it { is_expected.to contain_keystone_config('database/max_pool_size').with_value('21') }
       it { is_expected.to contain_keystone_config('database/max_overflow').with_value('21') }
       it { is_expected.to contain_keystone_config('database/retry_interval').with_value('11') }
-      it { is_expected.to contain_package('keystone-backend-package').with({ :ensure => 'present', :name => platform_params[:pymysql_package_name] }) }
-
     end
 
     context 'with MySQL-python library as backend package' do
@@ -83,11 +81,21 @@ describe 'keystone::db' do
       }
     end
 
-    let :platform_params do
-      { :pymysql_package_name => 'python-pymysql' }
-    end
-
     it_configures 'keystone::db'
+
+    context 'using pymysql driver' do
+      let :params do
+        { :database_connection     => 'mysql+pymysql://keystone:keystone@localhost/keystone', }
+      end
+
+      it 'install the proper backend package' do
+        is_expected.to contain_package('keystone-backend-package').with(
+          :ensure => 'present',
+          :name   => 'python-pymysql',
+          :tag    => 'openstack'
+        )
+      end
+    end
   end
 
   context 'on Redhat platforms' do
@@ -97,11 +105,14 @@ describe 'keystone::db' do
       }
     end
 
-    let :platform_params do
-      { :pymysql_package_name => 'python2-PyMySQL' }
-    end
-
     it_configures 'keystone::db'
+
+    context 'using pymysql driver' do
+      let :params do
+        { :database_connection     => 'mysql+pymysql://keystone:keystone@localhost/keystone', }
+      end
+      it { is_expected.not_to contain_package('keystone-backend-package') }
+    end
   end
 
 end
